Verifies signatures in CI

emkll opened the pull request:
very simple scripts that iterates through files in the workstation repo and verifies their signature. Specifies `--test` in this repo as this is the dev repo signed with the test key.

Note that the repo as initially cloned does not have git lfs hooks enabled due to circle re-writing the git urls per the gitconfig. We reclone via HTTPS inside the checked out repo to avoid having to deal with rewriting remote urls in the repo git config in the circle container.

### Test plan
- [ ] Should we even do this? Every clone is 3GB (and growing), will this pose an issue for our LFS bandwith quota?
- [ ] Is ci passing on this branch?
- [ ] The public keys are correct
- [ ] Verify that we are running CI per-PR (on push, to reduce git-lfs bandwidth)
- [ ] Is ci failing on the branch where i have commited an unsigned rpm ? https://circleci.com/gh/freedomofpress/securedrop-workstation-dev-rpm-packages-lfs/8

conorsch commented:
Looking good. Haven't done an in-depth functional review, since I'd prefer to shake out RPM nightly implications such as https://github.com/freedomofpress/securedrop-workstation/issues/406#issuecomment-578366922 before deciding what the sig verification story needs to look like.

> Every clone is 3GB

On a full copy of this repo, I'm seeing `5207MB`. However, if I clone with `--depth=1`, I see `1627MB`. So perhaps we should consider shallow clones in CI, since we only care about verifying sigs on _new_ artifacts.

emkll commented:
Thanks @conorsch , updated with shallow clone and rebased on latest master. I've also opened https://github.com/freedomofpress/securedrop-debian-packaging/pull/135 to reduce lfs usage for nightlies as well.

conorsch commented on scripts/check.py:
```diff
@@ -0,0 +1,75 @@
+        # find a way other than parsing stdout
+        print(str(output))
+        if "digests signatures OK" not in str(output):
+            fail("Signture verification failed for {}:{}".format(path, output))
```
Nit: use `output.decode("utf-8")`, to clean up the formatting a bit. Right now, the tool prints bytes-formatted lines:

```
b'workstation/dom0/f25/qubes-template-securedrop-workstation-buster-4.0.1-201911061435.noarch.rpm: digests signatures OK\n'
b'workstation/dom0/f25/securedrop-workstation-dom0-config-0.1.1.dev.20200125.060922-1.fc25.noarch.rpm: digests signatures OK\n'
b'workstation/dom0/f25/securedrop-workstation-dom0-config-0.1.1.dev.20200126.060943-1.fc25.noarch.rpm: digests signatures OK\n'
b'workstation/dom0/f25/securedrop-workstation-dom0-config-0.1.1.dev.20200128.060927-1.fc25.noarch.rpm: digests signatures OK\n'
b'workstation/dom0/f25/securedrop-workstation-dom0-config-0.1.1.dev.20200127.061227-1.fc25.noarch.rpm: digests signatures OK\n'
```

If we trim the whitespace, we could strengthen the validation check to be `== "<path>: digests signatures OK"`.

conorsch commented on scripts/check.py:
```diff
@@ -0,0 +1,75 @@
+    parser.add_argument('--dev', action='store_true', default=False)
+    parser.add_argument('--verify', action='store_true', default=True)
+    parser.add_argument('--all', action='store_true', default=False)
+    parser.add_argument('packages', type=str, nargs='*', help='Files to sign/verify')
```
Out of scope for a CI-only script, but in order to guard against user error, we should handle the no-args invocation of the script:

```
$ ./scripts/check.py 
Namespace(all=False, dev=False, packages=[], verify=True)
rpm: RPM should not be used directly install RPM packages, use Alien instead!
rpm: However assuming you know what you are doing...
error: package gpg-pubkey is not installed
$ echo $?
0
```

We definitely want that to exit non-zero, no? Can't do `nargs='*'` -> `nargs='+'`, since `--all` means `packages` is optional, but an `if not args.all and len(packages) < 1: fail()` would help.

emkll commented on scripts/check.py:
```diff
@@ -0,0 +1,75 @@
+    try:
+        subprocess.check_call(["rpm", "--erase", "--allmatches", "gpg-pubkey"])
+    except subprocess.CalledProcessError:
+        pass
```
Good point, added a check to query rpm for gpgkeys, through this should not happen in CI due to fresh container for each build. No more `pass`

emkll commented:
@conorsch thanks for the review, all great points. All comments have been addressed, commit squashed, ready for re-review
